compiler/optimizer/OMROptimizationManager.cpp

```diff
@@ -49,7 +49,7 @@ OMR::OptimizationManager::OptimizationManager(TR::Optimizer *o, OptimizationFact
         _numPassesCompleted(0),
         _optData(NULL),
         _optPolicy(NULL),
-        _enabled(false),
+        _enabled(!o->comp()->isDisabled(optNum)),
```

Comment:
Agreed, opt A doesn't know (at least not easily) what opt B's opt index is when the optimizer calls B's perform so the regex opt index check has to happen in the optimizer dispatcher loop but the regex opt name check can still be added to isDisabled() if desired. IMO, regex opt index check can be a part of query asking whether an opt will run or not; whereas regex opt name check is a part of query asking whether an opt has been disabled or not (though that is not how the code has been working so far).

Please note that the query isDisabled() currently only returns true if a command-line option like disableFoo has been detected under the original assumption that all opts are enabled unless  instructed otherwise. So asking isDisabled() doesn't necessarily mean we are asking "the most correct question". It could actually be misleading today since a project using OMR may not even support opt Foo for instance. This was addressed with the new query implementation in this PR.